psp/models/regressors/decision_trees.py

```diff
@@ -58,8 +60,24 @@ def _prepare_features(self, features: BatchedFeatures) -> tuple[np.ndarray, list
         """
         # Ignore the features that start with an underscore.
         feature_names = list([n for n in features if not n.startswith("_")])
-        # Stack all the features together
-        new_features = np.stack([features[n] for n in feature_names], axis=-1)
+
+        if is_training:
+            # At train-time, we note the feature names.
+            # We make a copy because this list is modified later.
+            self._feature_names = list(feature_names)
+        else:
```

Comment:
will there be the case where you run something, and dont go thorugh the first `if` statement, i.e just run inference, and not train

Reply:
It wouldn't make sense no. If we get in a place where it's easy to use an untrained regressor by mistake then we can make a cleaner error, but I think we want it to crash.